TuneUp/TuneUpWindow.xaml.cs

```diff
@@ -23,46 +23,49 @@ public partial class TuneUpWindow : Window
 
         ViewLoadedParams viewLoadedParams;
 
+        ICommandExecutive commandExecutive;
 
+        /// <summary>
+        /// Create the TuneUp Window
+        /// </summary>
+        /// <param name="vlp"></param>
         public TuneUpWindow(ViewLoadedParams vlp)
         {
             InitializeComponent();
             dynamoViewModel = (vlp.DynamoWindow.DataContext as DynamoViewModel);
             viewLoadedParams = vlp;
-        }
-
-        private void DataGridRow_Selected(object sender, RoutedEventArgs e)
-        {
-            var nodeModel = ((ProfiledNodeViewModel)((DataGridRow)sender).DataContext).NodeModel;
-            nodeModel.Select();
-
-            /*var nodeViewModels = dynamoViewModel.CurrentSpaceViewModel.Nodes;
 
-            NodeViewModel nodeViewModel = null;
-            foreach(var n in nodeViewModels)
-            {
-                if (n.Id == nodeModel.GUID)
-                {
-                    nodeViewModel = n;
-                }
-            }
-
-            var nodePt = new Point2D(nodeModel.X, nodeModel.Y);
-
-            var zoomArgs = new ModelEventArgs(nodeModel, nodeModel.X, nodeModel.Y, true);
-
-            dynamoViewModel.CurrentSpaceViewModel.FitViewToNode(nodeViewModel);*/
+            commandExecutive = vlp.CommandExecutive;
         }
 
         private void NodeAnalysisTable_SelectionChanged(object sender, SelectionChangedEventArgs e)
         {
-            foreach (var row in e.AddedItems)
+            // Get NodeModel(s) that correspond to selected row(s)
+            var selectedNodes = new List<NodeModel>();
+            foreach(var item in e.AddedItems)
             {
-                (row as ProfiledNodeViewModel).NodeModel.Select();
+                selectedNodes.Add((item as ProfiledNodeViewModel).NodeModel);
             }
-            foreach (var row in e.RemovedItems)
+
+            if (selectedNodes.Count() > 0)
             {
-                (row as ProfiledNodeViewModel).NodeModel.Deselect();
+                // Select
+                var command = new DynamoModel.SelectModelCommand(selectedNodes.Select(nm => nm.GUID), ModifierKeys.None);
+                viewLoadedParams.CommandExecutive.ExecuteCommand(command, Guid.NewGuid().ToString(), "TuneUp");
+
+                // Focus on selected
+                var ce = viewLoadedParams.GetType().GetProperty("ViewModelCommandExecutive");
```

Comment:
can you rip this string out to a const and add a comment here saying why you are doing this using reflection for API compatibility.